messaging.rst

```diff
@@ -276,6 +276,148 @@ This should match `the Secret message in encoding/messages file <https://github.
++===============================+===============+================================================================+
+|  cmdid                        | one byte      | Value 15 (indicating ``Withdraw Request``)                     |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  chain identifier             | uint256       | See :ref:`balance-proof-offchain                               |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  channel identifier           | uint256       | See :ref:`balance-proof-offchain`                              |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  token network address        | address       | See :ref:`balance-proof-offchain`                              |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  message identifier           | uint64        | An ID used in ``Delivered`` and ``Processed`` acknowledgements |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  participant                  | address       | The address of the withdraw requesting node                    |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  total_withdraw               | uint256       | The new monotonic ``total_withdraw`` value                     |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  expiration                   | uint256       | The block number at which withdraw request is no longer        |
+|                               |               | usable on-chain.                                               |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  nonce                        | uint64        | See :ref:`balance-proof-offchain`                              |
++-------------------------------+---------------+----------------------------------------------------------------+
+|  signature                    | bytes         | Elliptic Curve 256k1 signature                                 |
++-------------------------------+---------------+----------------------------------------------------------------+
+
+.. _withdraw-confirmation-message:
+
+Withdraw Confirmation
+-------------
+
+Message used by the :ref:`withdraw-request-message` receiver to confirm the request after validating it's input.
+
+Preconditions
+^^^^^^^^^^^^^
+
+- The channel for which the withdraw is confirmed should be open.
+- The received confirmation should map to a previously sent request.
```

Comment:
How does the mapping work? Is the Withdraw Confirmation an envelope for Withdraw Request?

Reply:
No not really. This is implementation specific where for the "initiator" of the withdraw, a map of all requests should be kept so that the withdraw confirmation should confirm to an existing key in the map.